zebra-consensus/src/transaction.rs

```diff
@@ -366,29 +366,29 @@ where
 
         let shielded_sighash = transaction.sighash(upgrade, HashType::ALL, None);
 
-        let _async_checks = Self::verify_transparent_inputs_and_outputs(
-            &request,
-            network,
-            inputs,
-            script_verifier,
-        )?
-        .and(Self::verify_sapling_shielded_data(
-            sapling_shielded_data,
-            &shielded_sighash,
-        )?)
-        .and(Self::verify_orchard_shielded_data(
-            orchard_shielded_data,
-            &shielded_sighash,
-        )?);
+        Ok(
+            Self::verify_transparent_inputs_and_outputs(
+                &request,
+                network,
+                inputs,
+                script_verifier,
+            )?
+            .and(Self::verify_sapling_shielded_data(
+                sapling_shielded_data,
+                &shielded_sighash,
+            )?)
+            .and(Self::verify_orchard_shielded_data(
+                orchard_shielded_data,
+                &shielded_sighash,
+            )?),
+        )
 
         // TODO:
         // - verify orchard shielded pool (ZIP-224) (#2105)
         // - ZIP-216 (#1798)
         // - ZIP-244 (#1874)
         // - remaining consensus rules (#2379)
         // - remove `should_panic` from tests
-
-        unimplemented!("V5 transaction validation is not yet complete");
```

Comment:
I think this change will conflict with @conradoplg's PR #2802.

We should put these changes in whichever PR actually needs them to pass the tests.
(Or just merge this PR, and rebase PR #2802.)

Reply:
I'll rebase, should be simple